RoosterPlanner.Api/Program.cs

```diff
@@ -14,14 +18,26 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            CreateHostBuilder(args).Build().Run();
         }
 
-        public static IWebHostBuilder CreateWebHostBuilder(string[] args)
+        public static IHostBuilder CreateHostBuilder(string[] args)
         {
-            return WebHost.CreateDefaultBuilder(args)
-                .UseIISIntegration()
-                .UseStartup<Startup>();
+        return Host.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((context, config) =>
+                {
+                    var configuration = config.Build();
+                    var azureServiceTokenProvider = new AzureServiceTokenProvider();
+                    var keyVaultClient = new KeyVaultClient(
+                        new KeyVaultClient.AuthenticationCallback(
+                            azureServiceTokenProvider.KeyVaultTokenCallback));
+
+                    config.AddAzureKeyVault(
+                        $"https://{configuration["KeyVaultName"]}.vault.azure.net/",
+                        keyVaultClient,
+                        new DefaultKeyVaultSecretManager());
+                })
+                .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseIISIntegration().UseStartup<Startup>(); });
```

Comment:
De methode 'ConfigureWebHostDefaults' heeft volgens mij al 'UseIISIntegration()' in zich...